configure.ac

```diff
@@ -355,33 +355,31 @@ OMPI_CHECK_UCX([ucx],
 num_transports=""
 if test -n "$with_portals4" -a "$with_portals4" != "no" ; then
     num_transports="x$num_transports"
+    transport="portals4"
+    transport_ofi="no"
+    transport_ucx="no"
+    AC_DEFINE([USE_PORTALS4], [1], [Define if Portals 4 transport active])
 fi
 if test -n "$with_ofi" -a "$with_ofi" != "no" ; then
     num_transports="x$num_transports"
+    transport="ofi"
+    transport_portals="no"
+    transport_ucx="no"
+    AC_DEFINE([USE_OFI], [1], [Define if OFI transport active])
 fi
 if test -n "$with_ucx" -a "$with_ucx" != "no" ; then
     num_transports="x$num_transports"
+    transport="ucx"
+    transport_ofi="no"
+    transport_portals="no"
+    AC_DEFINE([USE_UCX], [1], [Define if UCX transport active])
+    AC_DEFINE([ENABLE_HARD_POLLING], [1], [Enable hard polling])
 fi
 
 if test -n "$num_transports" -a "$num_transports" != "x" ; then
     AC_MSG_ERROR([Cannot select more than one transport, see --help for details])
-fi
-
-if test -n "$with_portals4" -a "$with_portals4" != "no" ; then
-    transport="portals4"
-    AC_DEFINE([USE_PORTALS4], [1], [Define if Portals 4 transport active])
-elif test -n "$with_ofi" -a "$with_ofi" != "no" ; then
-    transport="ofi"
-    AC_DEFINE([USE_OFI], [1], [Define if OFI transport active])
-elif test -n "$with_ucx" -a "$with_ucx" != "no" ; then
-    transport="ucx"
-    AC_DEFINE([USE_UCX], [1], [Define if UCX transport active])
-    AC_DEFINE([ENABLE_HARD_POLLING], [1], [Enable hard polling])
 else
     transport="none"
-    transport_portals4="no"
-    transport_ofi="no"
-    transport_ucx="no"
     AC_MSG_WARN([No transport requested])
```

Comment:
It looks like the `else` case will now be triggered even when a transport is selected successfully.

Reply:
Yikes... thanks.

Reply:
Yeah, it should be an `else`. We can check if num_transports is empty and then we assign transport as none.

Reply:
@jdinan @wrrobin - Sorry about that... does the new version look correct?

Reply:
@davidozog Thanks. Looks good to me.

Reply:
I think this is ok. I have a slight preference for the old structure, which did the error check for multiple transports before setting up the transport, but I think this new code should also work.

Reply:
Yeah I had a similar thought after this last change - it's "6 of one half-dozen of the other" in my opinion.

I think I'll just revert to the first commit on this PR, unless @wrrobin has any qualms.

Reply:
I am ok either way. Current plan sounds good.